### PR TITLE
C#: Re-factor Xss to use the new data flow API.

### PR DESCRIPTION
**Pull request by michaelnebel**

### CODE REVIEW

**Comment by michaelnebel** [comment]
DCA looks good, except for a failure on mono (both variants failed - so most likely a spurious failure).

**Comment by michaelnebel** [comment]
After re-running DCA for the mono project (failed in the first attempt), we get an extra alert.
However, this is the same alert that disappeared during the last quarter: https://github.com/github/codeql-dca-main/issues/12334
Let us be happy that it is back again.

File: csharp/ql/lib/semmle/code/csharp/security/dataflow/XSSQuery.qll [hunk 1]
```diff
@@ -69,13 +67,14 @@ module PathGraph {
...
-    DataFlow2::PathGraph::subpaths(arg.asDataFlowNode(), par.asDataFlowNode(), ret.asDataFlowNode(),
-      out.asDataFlowNode())
+    XssTracking::PathGraph::subpaths(arg.asXssDataFlowNode(), par.asXssDataFlowNode(),
+      ret.asXssDataFlowNode(), out.asXssDataFlowNode())
   }
 }
 
 private newtype TXssNode =
-  TXssDataFlowNode(DataFlow2::PathNode node) or
+  TXssDataFlowNode(DataFlow2::PathNode node) or // Deprecated
```
Comment (hvitved) [inline csharp/ql/lib/semmle/code/csharp/security/dataflow/XSSQuery.qll:76]: I don't think we want this, as it means we basically have to compute the flow graph twice. Instead, I think for this case we should simply break backwards compatibility.

**Review by hvitved — CHANGES_REQUESTED** [review]
One comment, otherwise LGTM.

File: csharp/ql/lib/semmle/code/csharp/security/dataflow/XSSQuery.qll (hunk 1 above)
Comment (michaelnebel) [reply csharp/ql/lib/semmle/code/csharp/security/dataflow/XSSQuery.qll:76]: Ok - that is fair enough.
I will do that instead and add a change node (I suppose that is breaking change which requires bumping the major version number?)

File: csharp/ql/src/change-notes/2023-04-20-xss-query-tainttracking.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+category: majorAnalysis
+---
+* C#: Extending `TaintTrackingConfiguration` in `XSSQuery.qll` no longer affects query results.
```
Comment (hvitved) [inline csharp/ql/src/change-notes/2023-04-20-xss-query-tainttracking.md:4]: I don't think this change note is needed, TBH. It was never intended for 3rd parties to customize the query by overriding the configuration itself; instead, they could add new sources or sinks.

Comment (michaelnebel) [reply csharp/ql/src/change-notes/2023-04-20-xss-query-tainttracking.md:4]: Alright - I will go ahead and remove it.

**Review by hvitved — APPROVED** [review]

**Pull request merged**